### PR TITLE
chore: add integration tests for staking module inside `gaia-rs`

### PR DESCRIPTION
**Pull request by dmvict**

### CODE REVIEW

File: x/staking/src/traits.rs [hunk 1]
```diff
@@ -23,7 +23,7 @@ pub trait BankKeeper<SK: StoreKey, M: Module>: Clone + Send + Sync + 'static {
 
     fn get_all_balances<DB: Database, CTX: QueryableContext<DB, SK>>(
         &self,
-        ctx: &mut CTX, //TODO: why is this mutable?
```
Comment (NYBACHOK) [inline x/staking/src/traits.rs:26]: It was mutable 'cause we used concrete context, which always was mutable

File: gaia-rs/tests/staking.rs [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+    BondStatus, CommissionRatesRaw, CommissionRaw, DelegationResponse, Description,
+    RedelegationResponse, Validator,
+};
+use std::{path::PathBuf, str::FromStr};
+use utilities::ACC_ADDRESS;
+
+#[path = "./utilities.rs"]
+mod utilities;
+
+fn run_tx_local(
```
Comment (NYBACHOK) [inline gaia-rs/tests/staking.rs:40]: Could we reuse this method in others tests?

File: gaia-rs/tests/staking.rs [hunk 3]
```diff
@@ -0,0 +1,490 @@
...
+            node: DEFAULT_TENDERMINT_RPC_ADDRESS.parse()?,
+            chain_id: ChainId::from_str("test-chain")?,
+            fee: None,
+            inner: WrappedGaiaTxCommands(command),
+        },
+        &GaiaCoreClient,
+    )
+}
+
+fn run_query_local(command: GaiaQueryCommands) -> anyhow::Result<GaiaQueryResponse> {
```
Comment (NYBACHOK) [inline gaia-rs/tests/staking.rs:61]: Same here

**Review by NYBACHOK — APPROVED** [review]

File: gaia-rs/tests/staking.rs (hunk 2 above)
Comment (dmvict) [reply gaia-rs/tests/staking.rs:40]: Do you mean `utilities.rs`? Of course, we can. I suggest  it's  a smaller abstraction that allows to reduce a lot of code. 

I've used it many times in the file.

Comment (NYBACHOK) [reply gaia-rs/tests/staking.rs:40]: May you use it in this PR? Or make another one

**Pull request merged**